docs/designs/3025-unmanaged-exit-codes.md

```diff
@@ -0,0 +1,123 @@
+<!-- The sections in this template are highly-suggested, but optional. They exist to
+provide an idea of what an in-depth proposal **might** include. If a different
+format fits your proposal, feel free to use it, with the expectation that:
+
+1. You must link to the proposal issue (github) under the title
+1. If you omit sections that are relevant, you may be asked by maintainers to
+   add them in order to best review the proposal. -->
+
+# Exit codes for `unmanaged-cluster` bootstrapping
+
+`unmanaged-cluster` bootstrapping should provide meaningful exit codes
+to capture success and failure.
+
+This will enable better interroperability with other tools
+and CI/CD systems.
+
+* [Proposal for exit codes in unmanaged-cluster](https://github.com/vmware-tanzu/community-edition/issues/3025)
+
+## Summary
+
+Currently, during `unmanaged-cluster` bootstrapping,
+whenever the command line tool exits control flow,
+it returns `0` back to the parent process.
+This means that `unmanaged-cluster` cannot provide meaningful
+feedback on error to a parent process
+without that parent parsing string output from `unmanaged-cluster`.
+
+Instead, we should provide meaningful codes
+that represent major steps during the bootstrapping process.
+This does not mean that every possible failure case needs an exit code.
+But rather, we should provide a subset of codes that communicate
+what step in the bootstrapping process failure occurred.
+
+## Motivation
+
+Without this change, our users won't be able to easily
+create automation around bootstrapping `unmanaged-cluster`s.
+Our users integrate tools like bash and python scripts or CI/CD systems
+into their bootstrapping workflows.
+
+### Goals
+
+* Provide a set of exit codes that meaninfully coordinate with failures during boostrapping
+* Capture exit codes for the most common failure states
+
+### Non-Goals/Future Work
+
+* Exit codes that capture every possible failure case
+* Recommendation on how to utilize the exit codes
+
+## Proposal
```

Comment:
I wanna know which package you wanna put this intelligence in.

I'd highly prefer `tanzu`, let's talk about it.